cli.go

```diff
@@ -742,6 +742,145 @@ func main() {
+						}
+					}
+
+					toBeFollowed := make([]string, 0)
+					if !noCheck { // exclude already-followed projects:
+						took := NewTimer()
+						Infof("Getting list of followed projects...")
+						projects, protoProjects, err := client.ListFollowedProjects()
+						if err != nil {
+							panic(err)
+						}
+						Infof("Currently %v projects (and %v proto) are followed; took %s", len(projects), len(protoProjects), took())
+
+						for _, repoURL := range repoURLs {
+							_, isFollowed := isAlreadyFollowedProject(projects, repoURL)
+							_, isFollowedProto := isAlreadyFollowedProto(protoProjects, repoURL)
+							isNOTFollowed := !isFollowed && !isFollowedProto
+							if isNOTFollowed {
+								toBeFollowed = append(toBeFollowed, repoURL)
+							}
+						}
+						toBeFollowed = Deduplicate(toBeFollowed)
+					} else {
+						toBeFollowed = Deduplicate(repoURLs)
+						repoNames = Deduplicate(repoNames)
+					}
+
+					totalToBeFollowed := len(toBeFollowed)
+					Infof("Will follow %v projects...", totalToBeFollowed)
+					if !force {
+						CLIMustConfirmYes("Do you want to continue?")
+					}
+
+					// Write toBeFollowed to temp file:
+					saveTargetListToTempFile("follow-by-code-search", toBeFollowed)
+
+					followedNew := 0
+
+					etac := eta.New(int64(totalToBeFollowed))
+
+					// follow repos:
+					for _, repoURL := range toBeFollowed {
+						envelope := follower(repoURL, etac)
+						if envelope != nil {
+							// if the project was NOT already known to lgtm.com,
+							// sleep to avoid triggering too many new builds:
+							isNew := !envelope.IsKnown()
+							if isNew {
+								followedNew++
+								// sleep:
+								time.Sleep(waitDuration)
+							}
+						}
+					}
+					Successf("Followed %v projects (%v new)", totalToBeFollowed, followedNew)
+
+					// add them to list
+					totalRepoNames := len(repoNames)
+					for len(repoNames) != 0 {
+						for repoNameIndex, repoName := range repoNames {
+							for _, projectListKey := range projectListKeys {
+								success, err := client.AddProjectToSelectionFromRepoName(projectListKey, repoName)
+								if err != nil {
+									panic(err)
+								}
+								if !success {
+									Warnf("Retrying later %s...", repoName) // Repo is new and hasn't been analysed yet.
+									break
+								} else {
+									Successf("Added %s to %s.", repoName, projectListKey)
+									repoNames = append(repoNames[:repoNameIndex], repoNames[repoNameIndex+1:]...) // https://www.delftstack.com/howto/go/how-to-delete-an-element-from-a-slice-in-golang/ (We need to preserve order)
+								}
+							}
+						}
+					}
+					Successf("Added %v projects to provided projects' keys", totalRepoNames)
+
+					return nil
+				},
+				Flags: []cli.Flag{
+					&cli.IntFlag{
+						Name:  "limit",
+						Usage: "Max number of code results.",
+					},
+					&cli.BoolFlag{
+						Name:  "force, y",
+						Usage: "Don't ask for confirmation.",
+					},
+					&cli.BoolFlag{
```

Comment:
New bool flag to avoid checking for followed repositories https://github.com/github/codeql/issues/4896